src/main/scala/cromwell/binding/WdlBinding.scala

```diff
@@ -115,6 +114,11 @@ object WdlBinding {
     }
   }
 
+  /* All MemberAccess ASTs that are not contained in other MemberAccess ASTs */
+  def findTopLevelMemberAccesses(expr: AstNode) = WdlBinding.findAstsWithTrail(expr, "MemberAccess").filter {
+    case (k, v) => !(v map {case a:Ast => a.getName} contains "MemberAccess")
```

Comment:
I think you'd be very hard-pressed to figure out how to make this a bunch of one-liners.  I'll do my best when I refactor it to make it as idiomatic as possible.  I also don't really agree that most things ought to be just a one-liner.  There's a cost that comes with that too.  Being hard-lined about it means you end up making a LOT of methods to do little things, even if it is `filterOutAstPathAncestorContains` that's only ever called in once place.  I find any uses of `map` or `filter` with any number of lines that I've seen in our code so far is perfectly readable.  Sure, I agree that they should always be as compact as possible, though a 10 line closure to a `map` is well within our ability to comprehend, right?